ambry-api/src/main/java/com/github/ambry/config/CompressionConfig.java

```diff
@@ -21,8 +21,8 @@
 public class CompressionConfig {
 
   // Boolean whether compression is enabled in PUT operation.
-  static final String COMPRESSION_ENABLED = "router.compression.enabled";
-  static final boolean DEFAULT_COMPRESSION_ENABLED = true;
+  public static final String COMPRESSION_ENABLED = "router.compression.enabled";
+  static final boolean DEFAULT_COMPRESSION_ENABLED = false;
```

Comment:
I think we can leave compression enabled.  The code changes below fixes decompression.  The issue is in decompression, not in the compression.  I suggest we enable compression so that we can test the decompression fix below.

Reply:
Par discussion, we can leave the feature off by default.  When deploying this feature, we need to create RB to enable this feature in each cluster and each fabric as deployment progresses.

Reply:
The rest LGTM